pkgs/development/python-modules/cwcwidth/default.nix

```diff
@@ -0,0 +1,20 @@
+{ lib, buildPythonPackage, fetchPypi, cython }:
+
+buildPythonPackage rec {
+  pname = "cwcwidth";
+  version = "0.1.4";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1azrphpkcyggg38xvkfb9dpc4xmmm90p02kf8dkqd4d6j5w96aj8";
+  };
+
+  nativeBuildInputs = [ cython ];
+
```

Comment:
```suggestion

  pythonImportsCheck = [ "cwcwidth" ];
  checkInputs = [ pytestCheckHook ];
```
Includes tests.

Reply:
I get:
```
============================= test session starts ==============================
platform linux -- Python 3.8.9, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /build/cwcwidth-0.1.4
collected 0 items / 1 error                                                    

==================================== ERRORS ====================================
___________________ ERROR collecting tests/test_cwcwidth.py ____________________
ImportError while importing test module '/build/cwcwidth-0.1.4/tests/test_cwcwidth.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/q6gfck5czr67090pwm53xrdyhpg6bx67-python3-3.8.9/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_cwcwidth.py:23: in <module>
    from cwcwidth import wcwidth, wcswidth
cwcwidth/__init__.py:32: in <module>
    from ._impl import wcwidth, wcswidth
E   ModuleNotFoundError: No module named 'cwcwidth._impl'
=========================== short test summary info ============================
ERROR tests/test_cwcwidth.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.14s ===============================
```

If I don't add the hook, tests pass:
```
----------------------------------------------------------------------
Ran 11 tests in 0.001s
```

Additionally, adding "cwcwidth._impl" to `pythonImportsCheck` works.

Reply:
The test failure is probably due to the same issue with not finding the properly cythonized code as in https://github.com/NixOS/nixpkgs/pull/82410#issuecomment-826402669 (see my workaround in https://github.com/NixOS/nixpkgs/pull/82410#issuecomment-827186298).

I try to avoid the default test command (i.e. the ``setuptoolsCheckPhase``) b/c that runs effectively ``python setup.py test``, which is being deprecated https://github.com/pypa/setuptools/issues/1684. If you use ``format="pyproject"``, then you're effectively forced to use ``pytestCheckHook`` (or a custom ``checkPhase``) b/c the tests aren't auto-run: https://github.com/NixOS/nixpkgs/blob/282473158cf8fafb736ed2a09f328be8b7ed2efa/pkgs/development/interpreters/python/mk-python-derivation.nix#L155-L163

Reply:
Done, I included the workaround